tests/product/image_builder.py

```diff
@@ -0,0 +1,203 @@
+from subprocess import check_output
+from sys import argv, exit
+from tests.configurable_cluster import ConfigurableCluster
+from tests.docker_cluster import DockerCluster, DockerClusterException
+from tests.no_hadoop_bare_image_provider import NoHadoopBareImageProvider
+
+from tests.product.mode_installers import StandaloneModeInstaller, \
+    YarnSliderModeInstaller
+from tests.product.prestoadmin_installer import PrestoadminInstaller
+from tests.product.standalone.presto_installer import StandalonePrestoInstaller
+from tests.product.topology_installer import TopologyInstaller
+from tests.product.yarn_slider.slider_installer import SliderInstaller
+
+import argparse
+
+
+class ProductTestImageBuilder:
+    STANDALONE_BARE_CLUSTER = 'bare'
+    BARE_CLUSTER = 'bare'
+    PA_ONLY_CLUSTER = 'pa_only_standalone'
+    STANDALONE_PRESTO_CLUSTER = 'presto'
+
+    PA_ONLY_YS_CLUSTER = 'pa_only_ys'
+    PA_SLIDER_CLUSTER = 'pa_slider'
+
+    _cluster_types = {
+        BARE_CLUSTER: [],
+        PA_ONLY_CLUSTER: [PrestoadminInstaller,
+                          StandaloneModeInstaller],
+        STANDALONE_PRESTO_CLUSTER: [PrestoadminInstaller,
+                                    StandaloneModeInstaller,
+                                    TopologyInstaller,
+                                    StandalonePrestoInstaller],
+        PA_ONLY_YS_CLUSTER: [PrestoadminInstaller,
+                             YarnSliderModeInstaller],
+        PA_SLIDER_CLUSTER: [PrestoadminInstaller,
+                            YarnSliderModeInstaller,
+                            SliderInstaller]
+    }
+
+    def __init__(self):
+        self.maxDiff = None
+        self.cluster = None
+        self.default_keywords = {}
```

Comment:
for initialization, I can remove maxDiff.
I could also remove self.cluster since it should be set by setup_cluster()

default_keywords:

    def _update_replacement_keywords(self, installers):
        for installer in installers:
            installer_instance = installer(self)
            self.default_keywords.update(installer_instance.get_keywords())

    def _setup_cluster(self, bare_image_provider, cluster_type):
    ...
                if not bare_cluster:
                    self._apply_post_install_hooks(installers)
                    self._update_replacement_keywords(installers)